lib/entry.js

```diff
@@ -209,4 +136,74 @@ Ep.runModuleSetters = function (module) {
   }
 };
 
+function call(setter, name, value, callback) {
+  if (name === "__esModule") {
+    // Ignore setters asking for module.exports.__esModule.
+    return;
+  }
+
+  if (value !== value) {
+    value = NAN;
+  } else if (typeof value === "undefined") {
+    value = UNDEFINED;
+  }
+
+  setter.last = setter.last || Object.create(null);
+
+  if (setter.last[name] !== value) {
```

Comment:
I think we should update `setter.last[name] = value` here instead of in the callback, unless there's a good reason to split that apart.